pkg/trie/triedb/node.go

```diff
@@ -26,11 +30,51 @@ type (
 	}
 
 	newValueRef struct {
-		hash *common.Hash
+		hash common.Hash
 		Data []byte
 	}
 )
 
+func NewEncodedValue(value nodeValue, partial []byte, childF childFunc) (codec.EncodedValue, error) {
+	var newHashedValueHash common.Hash
+	switch v := value.(type) {
+	case newValueRef:
```

Comment:
question: the `newValueRef` can be a value that a node in the trie holds? or is this a ref only to hash of child nodes?